PhysicsTools/PatAlgos/plugins/PATJetUpdater.cc

```diff
@@ -123,11 +123,11 @@ void PATJetUpdater::produce(edm::Event & iEvent, const edm::EventSetup & iSetup)
 
   for (edm::View<reco::Jet>::const_iterator itJet = jets->begin(); itJet != jets->end(); itJet++) {
 
-    // construct the Jet from the ref -> save ref to original object
+    // clone the input Jet to save ref to the original jet it was derived from
     unsigned int idx = itJet - jets->begin();
     edm::RefToBase<reco::Jet> jetRef = jets->refAt(idx);
-    edm::Ptr<reco::Jet> jetPtr = jets->ptrAt(idx);
-    Jet ajet( edm::RefToBase<Jet>(jetRef.castTo<JetRef>()) );
+    const Jet * origJet = dynamic_cast<const Jet *>( jetRef.get() );
+    Jet ajet( *origJet );
```

Comment:
@mverzett: If a pat::Jets is constructed from an object reference, then this newly created pat::Jet will be pointing to this object from which it was constructed. This is quite bad when using the updateJetCollection() function because every updated pat::Jet will lose memory of the original object it was derived from. This is particularly relevant for fat jets and subjets. If you decided to update either fat jets or subjets, you will no longer be able to figure out which subjet belongs to which fat jet.

@ahinzmann: Yes.